internal/driver/driver_test.go

```diff
@@ -1,56 +1,265 @@
+// 					Attributes: map[string]interface{}{
+// 						"getFunction": "GetDeviceInformation",
+// 						"service":     "Device",
+// 					},
+// 					Type: "Object",
+// 				}},
+// 			getFunction: "GetDeviceInformation",
+// 			data:        "",
+// 			expected: []*sdkModel.CommandValue{
+// 				{
+// 					DeviceResourceName: "DeviceInformation",
+// 					Type:               "Object",
+// 					Value: device.GetDeviceInformationResponse{
+// 						Manufacturer:    "Intel",
+// 						Model:           "SimCamera",
+// 						FirmwareVersion: "2.5a",
+// 						SerialNumber:    "9a32410c",
+// 						HardwareId:      "1.0",
+// 					},
+// 					Origin: 0,
+// 					Tags:   nil,
+// 				}},
+// 		},
+// 	}
+// 	for _, test := range tests {
+// 		test := test
+// 		t.Run(test.name, func(t *testing.T) {
+// 			d := &Driver{}
+// 			mockCallOnvifFunction := &mockCallOnvifFunction{}
+// 			mockGetOnvifClient := &mockGetOnvifClient{}
+// 			for _, req := range test.reqs {
+// 				mockCallOnvifFunction.On("CallOnvifFunction", req, test.getFunction, []byte(test.data)).
+// 					Return()
+// 			}
+// 			mockGetOnvifClient.GetOnvifClient(test.deviceName)
+// 			actual, err := d.HandleReadCommands(test.deviceName, test.protocols, test.reqs)
+// 			if test.errorExpected {
+// 				require.Error(t, err)
+// 			}
+// 			assert.Equal(t, test.expected, actual)
+// 		})
+// 	}
+// }
+
+// TestUpdateDevice: test for the proper updating of device information
+func TestUpdateDevice(t *testing.T) {
+	driver, mockService := createDriverWithMockService()
+	tests := []struct {
+		device  models.Device
+		devInfo *device.GetDeviceInformationResponse
+
+		expectedDevice models.Device
+		errorExpected  bool
+	}{
+		{
+			device: contract.Device{
+				Name: "testName",
+			},
+			devInfo: &device.GetDeviceInformationResponse{
+				Manufacturer:    "Intel",
+				Model:           "SimCamera",
+				FirmwareVersion: "2.5a",
+				SerialNumber:    "9a32410c",
+				HardwareId:      "1.0",
+			},
+			errorExpected: false,
+		},
+		{
+			device: contract.Device{
+				Name: "unknown_unknown_device",
+			},
+			devInfo: &device.GetDeviceInformationResponse{
+				Manufacturer:    "Intel",
+				Model:           "SimCamera",
+				FirmwareVersion: "2.5a",
+				SerialNumber:    "9a32410c",
+				HardwareId:      "1.0",
+			},
+			expectedDevice: contract.Device{
+				Name: "Intel-SimCamera-",
+			},
+			errorExpected: false,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.device.Name, func(t *testing.T) {
+			mockService.On("RemoveDeviceByName", test.device.Name).Return(nil).Once()
+			mockService.On("AddDevice", test.expectedDevice).Return(test.expectedDevice.Name, nil).Once()
+			mockService.On("UpdateDevice", test.device).Return(nil).Once()
+
+			err := driver.updateDevice(test.device, test.devInfo)
+
+			if test.errorExpected {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestDriver_RemoveDevice(t *testing.T) {

```

Comment:
Add comment and fix the naming by removing `underscore` to be consistent with other unit tests.

Reply:
like the other func, I think the name is fine.

Reply:
Is this an acceptable comment?

Reply:
I would write:
```go
// TestDriver_RemoveDevice tests the different code flows of the RemoveDevice when called with an actual device
// versus when called with the control plane device.
```

Reply:
Comment added